src/parallel/pipeline_executor.cpp

```diff
@@ -89,20 +90,33 @@ bool PipelineExecutor::TryFlushCachingOperators() {
 			finalize_result = OperatorFinalizeResultType::HAVE_MORE_OUTPUT;
 		}
 
-		push_result = ExecutePushInternal(curr_chunk, flushing_idx + 1);
+		auto push_result = ExecutePushInternal(curr_chunk, chunk_budget, flushing_idx + 1);
 
 		if (finalize_result == OperatorFinalizeResultType::HAVE_MORE_OUTPUT) {
 			should_flush_current_idx = true;
 		} else {
 			should_flush_current_idx = false;
 		}
 
-		if (push_result == OperatorResultType::BLOCKED) {
+		switch (push_result) {
+		case OperatorResultType::BLOCKED: {
 			remaining_sink_chunk = true;
 			return false;
-		} else if (push_result == OperatorResultType::FINISHED) {
+		}
+		case OperatorResultType::HAVE_MORE_OUTPUT: {
+			// The chunk budget was used up, pushing the chunk through the pipeline created more chunks
+			// we need to continue this the next time Execute is called.
```

Comment:
Can we assert that the chunk budget was depleted here using the method I proposed above? I'd like to be sure that this comment is correct